src/program/lwaftr/run_nohw/run_nohw.lua

```diff
@@ -73,24 +77,20 @@ function run(parameters)
    config.link(c, "b4if.tx -> aftr.v6")
    config.link(c, "aftr.v4 -> inet.rx")
    config.link(c, "aftr.v6 -> b4if.rx")
+   config.link(c, "aftr.hairpin_out -> aftr.hairpin_in")
+
+   engine.configure(c)
 
-   if verbosity >= 1 then
-      local csv = CSVStatsTimer:new(bench_file)
+   if opts.verbosity then
+      local csv = CSVStatsTimer:new(opts.bench_file)
       csv:add_app("inet", {"tx", "rx"}, { tx = "IPv4 TX", rx = "IPv4 RX" })
-      csv:add_app("tob4", {"tx", "rx"}, { tx = "IPv6 TX", rx = "IPv6 RX" })
+      csv:add_app("b4if", {"tx", "rx"}, { tx = "IPv6 TX", rx = "IPv6 RX" })
       csv:activate()
-
-      if verbosity >= 2 then
-         timer.activate(timer.new("report", function ()
-            app.report_apps()
-         end, 1e9, "repeating"))
-      end
    end
 
-   engine.configure(c)
-   engine.main {
-      report = {
-         showlinks = true;
-      }
-   }
+   if opts.duration then
+      engine.main({duration = opts.duration, report = { showlinks = true }})
+   else
+      engine.main()
```

Comment:
Links report were not shown before, because the program run indefinitely by default. The point is that unless a duration is set in `engine.main()` the program never finishes. If the program never finishes, reports are never shown.

The bottom line is to show a links report or an apps report, it's necessary to set a duration. If duration is not set just simply run the program indefinitely without setting any report option.